tests/kafkatest/tests/core/replica_scale_test.py

```diff
@@ -0,0 +1,139 @@
+                                                    "numPartitions": partition_count, "replicationFactor": replication_factor
+                                                }})
+        produce_workload = trogdor.create_task("replicas-produce-workload", produce_spec)
+        produce_workload.wait_for_done(timeout_sec=600)
+        self.logger.info("Completed produce bench")
+
+        consume_spec = ConsumeBenchWorkloadSpec(0, TaskSpec.MAX_DURATION_MS,
+                                                consumer_workload_service.consumer_node,
+                                                consumer_workload_service.bootstrap_servers,
+                                                target_messages_per_sec=10000,
+                                                max_messages=3400000,
+                                                consumer_conf={},
+                                                admin_client_conf={},
+                                                common_client_conf={},
+                                                active_topics=["replicas_produce_consume_[0-2]"])
+        consume_workload = trogdor.create_task("replicas-consume-workload", consume_spec)
+        consume_workload.wait_for_done(timeout_sec=600)
+        self.logger.info("Completed consume bench")
+
+        trogdor.stop()
+
+    @cluster(num_nodes=12)
+    @parametrize(topic_count=500, partition_count=34, replication_factor=3)
+    def test_clean_bounce(self, topic_count, partition_count, replication_factor):
+        topics_create_start_time = time.time()
+        for i in range(topic_count):
+            topic = "topic-%04d" % i
+            print("Creating topic %s" % topic)  # Force some stdout for Jenkins
+            topic_cfg = {
+                "topic": topic,
+                "partitions": partition_count,
+                "replication-factor": replication_factor,
+                "configs": {"min.insync.replicas": 1}
+            }
+            self.kafka.create_topic(topic_cfg)
+        topics_create_end_time = time.time()
+        self.logger.info("Time to create topics: %d" % (topics_create_end_time - topics_create_start_time))
+
+        restart_times = []
+        for node in self.kafka.nodes:
+            broker_bounce_start_time = time.time()
+            self.kafka.stop_node(node, clean_shutdown=True, timeout_sec=600)
+            self.kafka.start_node(node, timeout_sec=600)
+            broker_bounce_end_time = time.time()
+            restart_times.append(broker_bounce_end_time - broker_bounce_start_time)
+            self.logger.info("Time to restart %s: %d" % (node.name, broker_bounce_end_time - broker_bounce_start_time))
+
+        self.logger.info("Restart times: %s" % restart_times)
+
+        delete_start_time = time.time()
+        for i in range(topic_count):
+            topic = "topic-%04d" % i
+            self.logger.info("Deleting topic %s" % topic)
+            self.kafka.delete_topic(topic)
```

Comment:
Adding batch deletion here would be useful also. This has caused problems in Kafka previously.

Reply:
Yes, definitely. Exposing the batch create/delete through `kafka-topics.sh` would be nice and help out a lot in the tests.